### ISSUE TITLE
Telemetry warns with if external cluster is configured

### ISSUE THREAD

**Issue by parth-gr** [rook/rook#12250]
**Is this a bug report or feature request?**
* Bug Report

```
2023-05-17 16:25:54.860417 W | telemetry: failed to set telemetry key "rook/cluster/mon/allow-multiple-per-node". failed to set "rook/cluster/mon/allow-multiple-per-node" in the mon config-key store. output: Error EACCES: access denied: exit status 13
2023-05-17 16:25:59.715808 W | telemetry: failed to set telemetry key "rook/cluster/mon/pvc/enabled". failed to set "rook/cluster/mon/pvc/enabled" in the mon config-key store. output: Error EACCES: access denied: exit status 13
2023-05-17 16:26:04.843972 W | telemetry: failed to set telemetry key "rook/cluster/mon/stretch/enabled". failed to set "rook/cluster/mon/stretch/enabled" in the mon config-key store. output: Error EACCES: access denied: exit status 13
2023-05-17 16:26:09.715949 W | telemetry: failed to set telemetry key "rook/cluster/storage/device-set/count/total". failed to set "rook/cluster/storage/device-set/count/total" in the mon config-key store. output: Error EACCES: access denied: exit status 13
2023-05-17 16:26:14.757679 W | telemetry: failed to set telemetry key "rook/cluster/storage/device-set/count/portable". failed to set "rook/cluster/storage/device-set/count/portable" in the mon config-key store. output: Error EACCES: access denied: exit status 13
2023-05-17 16:26:19.715026 W | telemetry: failed to set telemetry key "rook/cluster/storage/device-set/count/non-portable". failed to set "rook/cluster/storage/device-set/count/non-portable" in the mon config-key store. output: Error EACCES: access denied: exit status 13

```

**Deviation from expected behavior:**

**Expected behavior:**

**How to reproduce it (minimal and precise):**

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary

**Logs to submit**:

* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

  To get logs, use `kubectl -n <namespace> logs <pod name>`
  When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
  Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Cluster Status to submit**:

* Output of krew commands, if necessary

  To get the health of the cluster, use `kubectl rook-ceph health`
  To get the status of the cluster, use `kubectl rook-ceph ceph status`
  For more details, see the [Rook Krew Plugin](https://rook.io/docs/rook/latest-release/Troubleshooting/krew-plugin)

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)):

**Comment by travisn** [comment]
Actually if it's an external cluster, rook shouldn't be reporting the telemetry. Let's disable the telemetry for external clusters. 
Only the provider cluster should be reporting telemetry. If the consumer cluster reports telemetry even if it has the admin keyring, it will show up in the ceph reports as belonging to the provider cluster. And if multiple consumers are connected to the same provider, the separate reports could just be conflicting.

**Comment by Nikhil-Ladha** [comment]
Hi, I am just trying to dig into rook and looking out for some `good-first-issues` to start with and stumbled upon this one. Can I give this a try? If yes, then to clarify the requirement,  it is just to not `[reportTelemetry](https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/cluster.go#L220)` if the cluster is external, right?

**Comment by parth-gr** [comment]
Yaa, you understand it right not to report it. You can open a PR to stop collecting the telemetry for now and also test it.
For the current file I already did the change https://github.com/rook/rook/pull/11493/files but you have to check from where else the package is reporting.

**Comment by parth-gr** [comment]
> Actually if it's an external cluster, rook shouldn't be reporting the telemetry. Let's disable the telemetry for external clusters. Only the provider cluster should be reporting telemetry. If the consumer cluster reports telemetry even if it has the admin keyring, it will show up in the ceph reports as belonging to the provider cluster. And if multiple consumers are connected to the same provider, the separate reports could just be conflicting.

@travisn a question,
> And if multiple consumers are connected to the same provider the 
Might be interesting in seeing what's the segregation of data between the different clusters, or maybe that can be just listed by the different pools in the provider.

**Comment by travisn** [comment]
> @travisn a question, And if multiple consumers are connected to the same provider the Might be interesting in seeing what's the segregation of data between the different clusters, or maybe that can be just listed by the different pools in the provider.

Agreed, if separate metrics are required per consumer cluster, they could have separate pools that are consumed by each consumer cluster.

**Comment by travisn** [comment]
I would expect this was already fixed by #11493. @parth-gr Does this still repro?

**Comment by Nikhil-Ladha** [comment]
> Yaa, you understand it right not to report it. You can open a PR to stop collecting the telemetry for now and also test it. For the current file I already did the change https://github.com/rook/rook/pull/11493/files but you have to check from where else the package is reporting.

I doubt the telemetry is still being reported, because I don't see any other reference to the `reportTelemetry` function throughout the codebase. Are you able to reproduce it @parth-gr ?

**Comment by parth-gr** [comment]
Hmm,
I don't remember it might be a 4.12 cluster and I see we don't have changes in 4.12 repo

**Comment by Nikhil-Ladha** [comment]
> Hmm, I don't remember it might be a 4.12 cluster and I see we don't have changes in 4.12 repo

In that case, a backport is only needed.
However, I am not sure how the backport for older z versions are considered.

**Comment by parth-gr** [comment]
Here the PR https://github.com/red-hat-storage/rook/pull/496

**Comment by travisn** [comment]
Already fixed by #11493
